src/commands/deployment/gate.ts

```diff
@@ -229,68 +235,85 @@ export class DeploymentGateCommand extends Command {
     api: APIHelper,
     evaluationId: string,
     timeoutSeconds: number
-  ): Promise<number> {
+  ): Promise<CommandResult> {
     this.logger.info('Waiting for gate evaluation results...')
 
     const maxWaitTime = timeoutSeconds * 1000
     let timePassed = Date.now() - this.startTime
+    let result: CommandResult | undefined
 
-    while (timePassed <= maxWaitTime) {
-      try {
-        const response = await api.getGateEvaluationResult(evaluationId)
-        const status = response.data.data.attributes.gate_status
-        const remainingTime = maxWaitTime - timePassed
-        const waitTime = Math.min(this.pollingInterval, remainingTime)
-        const waitTimeInSeconds = Math.floor(waitTime / 1000)
-
-        switch (status) {
-          case 'pass':
-            this.logger.info(chalk.green(`\t${ICONS.SUCCESS} Gate evaluation passed`))
-
-            return 0
-          case 'fail':
-            this.logger.info(chalk.red(`\t${ICONS.FAILED} Gate evaluation failed`))
-
-            return 1
-          case 'in_progress':
-            const rules = response.data.data.attributes.rules
-            const totalRules = rules.length
-            const completedRules = rules.filter((rule) => rule.status !== 'in_progress').length
-
-            this.logger.info(
-              `\tGate evaluation in progress (${completedRules}/${totalRules} rules completed). Retrying in ${waitTimeInSeconds}s...`
-            )
-            break
-
-          default:
-            this.logger.warn(`Unknown gate evaluation status: ${status}, retrying in ${waitTimeInSeconds}s...`)
-        }
-
-        await new Promise((resolve) => setTimeout(resolve, waitTime))
-        timePassed = Date.now() - this.startTime
-      } catch (error) {
-        this.logger.error(
-          `Error polling for gate evaluation results: ${error instanceof Error ? error.message : String(error)}`
-        )
+    while (timePassed < maxWaitTime) {
+      const remainingTime = maxWaitTime - timePassed
+      const waitTime = Math.min(this.pollingInterval, remainingTime)
+      const waitTimeInSeconds = Math.floor(waitTime / 1000)
 
-        return this.getExitCodeForDatadogError()
+      result = await this.getEvaluationResult(api, evaluationId)
+      if (result === 'PASS' || result === 'FAIL') {
```

Comment:
Should we do `if (result)` also here, similarly to below?

Reply:
True! Updated!